packages/svelte/src/internal/client/reactivity/effects.js

```diff
@@ -118,20 +118,29 @@ function create_effect(type, fn, sync) {
  * @returns {boolean}
  */
 export function effect_active() {
-	return current_effect ? (current_effect.f & (BRANCH_EFFECT | ROOT_EFFECT)) === 0 : false;
+	if (current_reaction && (current_reaction.f & DERIVED) !== 0) {
+		return (current_reaction.f & UNOWNED) === 0;
```

Comment:
Should this be?

```js
if ((current_reaction.f & UNOWNED) === 0) return true
```

Then it goes to the next block to check for an active current effect.

Reply:
I don't think we do, no.

As things stand today, if it did fall through to the current effect, and as a result `$effect.active()` returned true leading to a new effect being created, that new effect would be a child of the unowned derived. As such, it would never run — it would just sit there taking up memory until the derived was GC'd.

So we'd need a corresponding change inside `create_effect`:

```diff
diff --git a/packages/svelte/src/internal/client/reactivity/effects.js b/packages/svelte/src/internal/client/reactivity/effects.js
index 0ab8da808..13c93b33c 100644
--- a/packages/svelte/src/internal/client/reactivity/effects.js
+++ b/packages/svelte/src/internal/client/reactivity/effects.js
@@ -92,8 +92,14 @@ function create_effect(type, fn, sync) {
 		effect.component_function = dev_current_component_function;
 	}
 
-	if (current_reaction !== null && !is_root) {
-		push_effect(effect, current_reaction);
+	let reaction = current_reaction;
+
+	if (reaction !== null && (reaction.f & (DERIVED | UNOWNED)) === (DERIVED | UNOWNED)) {
+		reaction = current_effect;
+	}
+
+	if (reaction !== null && !is_root) {
+		push_effect(effect, reaction);
 	}
 
 	if (sync) {
@@ -119,7 +125,7 @@ function create_effect(type, fn, sync) {
  */
 export function effect_active() {
 	if (current_reaction && (current_reaction.f & DERIVED) !== 0) {
-		return (current_reaction.f & UNOWNED) === 0;
+		if ((current_reaction.f & UNOWNED) === 0) return true;
 	}
 
 	if (current_effect) {
```

With that, effects created inside unowned deriveds do run as long as the derived value is being read inside an effect, as in [this somewhat contrived example](https://svelte-5-preview-git-effect-active-derived-svelte.vercel.app/#H4sIAAAAAAAAE6VU226bQBD9lS3NAzSOafKIgarfUaoKs4OzyrKL9kJkIf69e-NiO-lFEbLxzjlzdubMyGPUEgoyyn6MEas7iLLoe99Hu0ide3uQA1AF5iy5Fo2N5LIRpFdlxSpFQaGGM0wU4QwV6E6qWkHc1lRCcjCMwOGvDLDBGwEG_9VwzRSI2FEcQbOZEiSSA_LprWaNU7_ORaOF5xpMcE3-6oU9hrk-UrAgBkEGwPvjOTbpRTkrVIq0KL6DtoVG7Wtz3WAKSFa4UgG8zrOP6V9yCnvKT_EYKpnmAuyzOUxLNLwFKC1YyPqCngJzms1bGMuVp7ld6wC6yJ92M0kupMGyZnuGDeW0WLMVCmaFOt3LfOXpOnOWH7VSZh6cNZQ0L8XoPdna4Ga591q7ebbfQiCZ3OqYaWI01FSDtHbkqZf9yxXLpn1aDl5OPvNXFI9rNPk3zXXxrvfL6_roQgsrdK2NyYCkOlMoqggT2dP6nKGTIPjgvh8UdCam4KHhVHdMZuixFfZj8Lo3J-iqyN1npUrvfv78VLpb89T8CnPzyNu9eNv9sO8L9DiVhJn6O2BqW-4iM342e786Nm9G3pfjRmrK0768xfwwt-DH0seUtK6EPA0W3LoRpnDjh2skgGsbfxz4f9j0rlG-nwvBi24v8XcN-6jMat0bLvq3-f_uOCYtARxlSmiYfk6_AeGoC1D6BQAA).

But there's a catch. If we read the derived _outside_ an effect, the child effect isn't created because `$effect.active()` is `false`. Meanwhile, it updates to the latest value. If we then read it _inside_ an effect, the value is already up to date, so the function doesn't re-run, and the child effect still isn't created. If you read the derived value inside an effect first, the child effect _is_ created. This feels like a sufficiently confusing difference in behaviour that we'd be much better off enforcing that unowned deriveds can't contain effects.

Interestingly: in the demo above (on this branch and also with the changes above), the effect inside the owned derived is created but doesn't run if the derived is read inside an effect. But it _does_ run if it's read outside an effect (try clicking 'read values' followed by 'increment'). That feels like a bug.

Reply:
Pushed that change to an `effect-active-derived-allow-unowned` branch for posterity